### PR TITLE
Rolling log file writer in new io module

### LINKED ISSUE #405: Feature request: log to files
Sometimes it is useful to log on Android and iOS to files, which can easily be zipped up and transferred for remote analysis.

Now that [kotlinx-io](https://github.com/Kotlin/kotlinx-io) has been released and supports multiplatform filesystem writes (via okio, and works at least for most platforms including Android and iOS), it would be great if Kermit would leverage this to provide filesystem writing capability out of the box, rather than having users roll their own `LogWriter`s.

For a logging tool, it would be useful also to support automatic roll-overs and deletion of older logs.

### PR DESCRIPTION
**Pull request by rocketraman**
A simple implementation to write request logs to files.

Implemented as a rolling system to avoid any one log file becoming too large, and setting an upper bound on device storage. The roll parameters are configurable.

This is useful for debugging user devices in production because apps can contain a feature to send these saved logs to a server for analysis.

Resolves #405 .

### CODE REVIEW

File: kermit-io/build.gradle.kts
Comment (rocketraman) [inline kermit-io/build.gradle.kts:40]: I removed JavaScript from this list, BUT I'm wondering if we should actually leave it. The code should work fine on Node.

**Comment by xxfast** [comment]
Great work 👍  Wondering how we can prioritise this the next release 🤔

**Comment by rocketraman** [comment]
> Great work 👍 Wondering how we can prioritise this for the next release 🤔

Thanks! You might want to post your interest on [the parent issue](https://github.com/touchlab/Kermit/issues/405), it might get more visibility there.

**Comment by KevinSchildhorn** [comment]
For my changes:
* Called `apiDump`
* Changed `libs.kotlinx.io` from implementation to api as you cannot add the library without them ( was going to do the same for `libs.kotlinx.datetime` but I want to prevent version mismatches as some people already use datetime)
* Re-ordered variables in `RollingFileLogWriter` so that the non-optional is first
* Added new `LogWriter` to sample app, and adding `mobileMain` sourceset to handle new `LogWriter`
* renaming `logPath` to `logFilePath` and `logPrefix` to `logFileName`

File: kermit-io/build.gradle.kts
Comment (rocketraman) [inline kermit-io/build.gradle.kts:65]: The reason I set it up as an api dependency is because `kotlinx.datetime.Clock` is exposed in the constructor.

Comment (rocketraman) [reply kermit-io/build.gradle.kts:65]: Oh I didn't do that... ok. I should have! I think this should still be an API dependency and we let Gradle do version conflict resolution.

Comment (KevinSchildhorn) [reply kermit-io/build.gradle.kts]: Ok, we'll keep it as `api` for now.

**Comment by KevinSchildhorn** [comment]
Very nice work, however we're running into `cinterop` issues when adding this change to the sample app, which causes a build error when testing. We want to ensure it works and builds properly before merging it in, so I'm holding off until we can fix this issue. The issue arises when building the sample app.

**Comment by rocketraman** [comment]
> Very nice work, however we're running into `cinterop` issues when adding this change to the sample app, which causes a build error when testing. We want to ensure it works and builds properly before merging it in, so I'm holding off until we can fix this issue.

How can I reproduce this? I've done `build` on the sample project without any errors locally.

**Review by KevinSchildhorn — APPROVED** [review]

**Comment by xxfast** [comment]
Are we handling the exceptions from this writer correctly? We've noticed there might be restrictions on ios writing to file when the device is locked, which causes this exception. I've opened #468 for this

```
kotlinx.io.IOException: fflush failed with errno 1 (Operation not permitted)
    at 0   debug.dylib                  0x11742031b        kfun:kotlinx.io.files.throwIOExceptionForErrno#internal + 547 
    at 1   debug.dylib                  0x11741fff3        kfun:kotlinx.io.files.FileSink#flush(){} + 239 
    at 2  debug.dylib                  0x11741415b        kfun:kotlinx.io.RealSink#flush(){} + 435 
    at 3   debug.dylib                  0x11767fc8f        kfun:co.touchlab.kermit.io.$writerCOROUTINE$0.invokeSuspend#internal + 1515  
```

**Pull request merged**
